algorithms/kernel/classifier/classifier_train.cpp

```diff
@@ -114,6 +135,22 @@ services::Status Input::checkImpl(const daal::algorithms::Parameter *parameter)
     data_management::NumericTablePtr labelsTable = get(labels);
     DAAL_CHECK_STATUS(s, data_management::checkNumericTable(labelsTable.get(), labelsStr(), 0, 0, 1, nRows));
 
+    // Check correctness of label table for interface 2 only
+    if (nClasses)
```

Comment:
it would be better to move this check under else if (algParameter2 != NULL)

Reply:
fixed